plugins/chat/spec/services/chat/auto_leave_channels_spec.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::AutoLeaveChannels do
+  describe ".call" do
+    subject(:result) { described_class.call(params: {}) }
+
+    before do
+      @previous_events = DiscourseEvent.events.dup
```

Comment:
We should avoid using instance variables (https://rspec.rubystyle.guide/#instance-variables). We should be able to simply use a `let!(:previous_events)` instead.

Reply:
Should we get warnings about those?

I'll have a try, thanks 👍

Reply:
There’s a cop for those, indeed: https://docs.rubocop.org/rubocop-rspec/cops_rspec.html#rspecinstancevariable. I’ll take a look at enabling it :slightly_smiling_face: